core/server/worker/src/main/java/alluxio/worker/netty/BlockReadHandler.java

```diff
@@ -182,6 +184,7 @@ private void openBlock(BlockReadRequestContext context, Channel channel) throws
             return;
           } catch (Exception e) {
             mWorker.closeUfsBlock(request.getSessionId(), request.getId());
+            context.setBlockReader(null);
```

Comment:
It is awkward to use reader as the flag of whether or not to close, since reader can be null here and we would still want to close

Reply:
talked offline. A TODO added